### ISSUE TITLE
Suggested simple enhancement for coloring the surface in mpl_toolkits/mplot3d/axes3d.py 

### ISSUE THREAD

**Issue by keithbriggs** [matplotlib/matplotlib#15426]
This line: 

> avg_z = np.array([ps[:, 2].mean() for ps in polys])

means that coloring the surface is only possibly with the z coordinate.

It would be very useful to allow an arbitrary function of ps here.

**Comment by timhoffm** [comment]
This is just the default. Use `facecolors` to supply arbitrary colors. See also:

https://matplotlib.org/gallery/mplot3d/custom_shaded_3d_surface.html#sphx-glr-gallery-mplot3d-custom-shaded-3d-surface-py

**Comment by QuLogic** [comment]
It _seems_ like there's nothing to be done here?

**Comment by ImportanceOfBeingErnest** [comment]
Maybe the point is, that unless you use `cstride=rstride=1`, you will not know the polygons in advance, so it might be hard to supply facecolors? Hence, it might be useful to allow for functions other than `.mean()`.

**Comment by keithbriggs** [comment]
Yes, if I remember correctly, there was this point, and also that you might want to color with ps[:,0], or some other function of ps.

**Comment by QuLogic** [comment]
OK, let's reopen for now. There is some chance that if performance is improved sufficiently, then `rstride` and `cstride` would no longer be needed.
